examples/transport/main.cpp

```diff
@@ -0,0 +1,161 @@
+#include <fmt/core.h>
+#include <graaflib/algorithm/graph_traversal.h>
+#include <graaflib/algorithm/shortest_path.h>
+#include <graaflib/io/dot.h>
+#include <graaflib/types.h>
+#include <graaflib/undirected_graph.h>
+
+struct station {
+  std::string name{};
+  double gaz_capacity{};
+};
+
+struct road {
+  int length{};
+  int get_weight() { return length; }
+};
```

Comment:
[minor]: I know this code works currently, but ideally the road struct should derive from `weighted_edge`  and override the `get_weight` method. This is a design flaw on the weighted edge design, which I plan to address in a separate PR. 

For now, I would do the following;

```suggestion
struct road : public weighted_edge<int> {
  int length{};
  [[nodiscard]] int get_weight() const noexcept override { return length; }
};
```